munkipkg

```diff
@@ -110,15 +110,22 @@ def validate_build_info_keys(build_info, file_path):
                 return None
 
 
-def read_build_info_plist(plist_path):
-    '''Reads and validates data in the build_info plist'''
+def read_build_info(path):
+    '''Reads and validates data in the build_info'''
     build_info = None
     try:
-        build_info = plistlib.readPlist(plist_path)
-    except ExpatError, err:
+        if path.endswith('.json'):
+            with open(path, 'r') as openfile:
+                build_info = json.load(openfile)
+        elif path.endswith(('.yaml', '.yml')):
+            with open(path, 'r') as openfile:
+                build_info = yaml.load(openfile)
+        elif path.endswith('.plist'):
+            build_info = plistlib.readPlist(path)
+    except (ExpatError, ValueError, yaml.scanner.ScannerError) as err:
         raise BuildError(
-            "%s is not a valid xml plist: %s" % (plist_path, str(err)))
-    validate_build_info_keys(build_info, plist_path)
+            "%s is not a valid %s file: %s" % (path, path.split('.')[-1], str(err)))
```

Comment:
One casualty of the consolidation is that the error is less clear if someone tries to use a binary plist. I don't think that's enough to not do this merge, though.